presto-main/src/main/java/io/prestosql/sql/planner/iterative/rule/TransformCorrelatedLateralJoinToJoin.java

```diff
@@ -61,10 +64,23 @@ public Result apply(LateralJoinNode lateralJoinNode, Captures captures, Context
                         decorrelatedNode.getNode(),
                         ImmutableList.of(),
                         lateralJoinNode.getOutputSymbols(),
-                        decorrelatedNode.getCorrelatedPredicates(),
+                        conjunctFilters(decorrelatedNode.getCorrelatedPredicates(), lateralJoinNode.getFilter()),
```

Comment:
Use `ExpressionUtils.combineConjuncts()`:

```java
ExpressionUtils.combineConjuncts(
    decorrelatedNode.getCorrelatedPredicates().orElse(TRUE_LITERAL), 
    lateralJoinNode.getFilter());
```

Reply:
Maybe `decorrelatedNode.getCorrelatedPredicates()` should return `TRUE_LITERAL` instead of `Optional.empty()`. I think having nested optional structure is not needed, because first we need to check if value is empty() and if it equal to `TRUE_LITERAL`.